arch/arc/core/cpu_idle.S

```diff
@@ -43,7 +43,18 @@ SECTION_FUNC(TEXT, z_arch_cpu_idle)
 
 	ld r1, [z_arc_cpu_sleep_mode]
 	or r1, r1, (1 << 4) /* set IRQ-enabled bit */
+/*
+ * It's found that (in nsim_hs_smp and hsdk), when cpu
+ * is sleeping, no response to inter-processor interrupt
+ * here is a workround
+ */
+#ifndef CONFIG_SMP
 	sleep r1
+#else
+	seti r1
+_z_arc_idle_loop:
+	b _z_arc_idle_loop
+#endif
```

Comment:
Alternatively, you could elaborate the spin loop that already exists in kernel/idle.c so it's predicated on more than just CONFIG_SCHED_IPI_SUPPORTED and use that one instead.  Though spinning is just two instructions, so maybe it doesn't matter.